app/src/main/java/com/medmapper/v33001/dto/Medicine.kt

```diff
@@ -3,16 +3,16 @@ package com.medmapper.v33001.dto
 import java.time.LocalDate
 
 data class Medicine(
-    val id: String = "",
-    val uid: String = "",
-    val name: String = "",
-    val quantity: Int = 0,
-    val prescriptionStrength: String = "",
-    val startDate: LocalDate = LocalDate.now(),
-    val prescriptionLength: String = "",
-    val time: Long = 0,
+    var id: String = "",
+    var uid: String = "",
+    var name: String = "",
+    var quantity: Int = 0,
+    var prescriptionStrength: String = "",
+    var startDate: LocalDate = LocalDate.now(),
+    var prescriptionLength: String = "",
+    var time: Long = 0,
     // Frequency per 24 hours
-    val frequency: Int = 0) {
+    var frequency: Int = 0) {
```

Comment:
Leave this as it were.  Assume variables are immutable (val), unless we have a very good reason for reassigning them.